tensorflow_privacy/privacy/membership_inference_attack/data_structures.py

```diff
@@ -374,6 +374,84 @@ def _append_array_shape(arr: np.array, arr_name: str, result):
     result.append(' %s with shape: %s,' % (arr_name, arr.shape))
 
 
+def _is_iterator(obj, obj_name):
+  """Checks whether obj is a generator."""
+  if obj is not None and not isinstance(obj, Iterator):
+    raise ValueError('%s should be a generator.' % obj_name)
+
+
+@dataclass
+class Seq2SeqAttackInputData:
```

Comment:
Copying this comment: "Now we have quite of bit of seq2sec-specific logic in files that were supposed to be generic (data_structures, models, membership_inference_attack), which makes those files quite heavy and much less generic than we hoped they'd be.

I wonder if this makes sense to put all seq2seq specific logic in a separate file (seq2seq.py), which will include pieces from the 3 files above.

WDYT?

This is food for thought, not blocking this PR on this discussion which might require a broader revamp of our file structure."

Reply:
Hmm, it would be a good idea to separate out the logic for seq2seq models since we could potentially add more attack types (like training the attacker on histograms of ranks).